### ISSUE TITLE
Failing to upload assignements through mod assign

### ISSUE THREAD

**Issue by JeryH** [jmcgettrick/MoodleDirectV2#132]
We know from using the TIN direct plugin that the API is unreliable because we quite regularly get calls from students unable to upload. In the case of the direct plugin the submission will fail and students are forced to try again later.

What we are seeing with the mod/assign integration is Moodle accepting the submission (files are in Moodle) but in some cases entire sets of submissions failing with a message "check API logs" (which we have done but don't know what the error codes mean).

My question at this point is what happens after that? It seems the submissions just sit there without any means to retry submission to TIN through its API?

If that is the case I think it would need a cron process to re-attempt upload of failures to TIN.

To add a little more to this a lecturer contacted me to say they had to abandon using TIN on mod assign as every assignment was failing to run through the API (other courses have one or two failures).

They hid that assignment and asked students to re-submit through another which I think was using TIN Direct.

I just went back into that same assignment and my submission was successful, so there were no issues with Moodle or the assignment, the only difference seemed to be the time at which I made my submission vs the student submissions which failed.

In short, without some process to re-attempt failed uploads to TIN I don't think the mod/assign integration is viable, at least not through submit.ac.uk during busy periods.
